doc/sphinx/docs/python/algorithms/py_algorithms.rst

```diff
@@ -3,78 +3,136 @@
+================================== ============================================ =============== ===========================================
+Common Name                        Name in PyGMO                                Type            Comments
+================================== ============================================ =============== ===========================================
+Monotonic Basin Hopping (MBH)      :class:`pygmo.mbh`                           S-CU            Exposed from C++
+Cstrs Self-Adaptive                :class:`pygmo.cstrs_self_adaptive`           S-C             Exposed from C++
+================================== ============================================ =============== ===========================================
+
+Local optimization 
+^^^^^^^^^^^^^^^^^^
+====================================================== ========================================= =============== =====================================================================
+Common Name                                             Name in PyGMO                             Type           Comments
+====================================================== ========================================= =============== =====================================================================
+Compass Search (CS)                                    :class:`pygmo.cs`                         S-CU            Exposed from C++
+COBYLA (from nlopt)                                    :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+BOBYQA (from nlopt)                                    :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+NEWUOA + bound constraints (from nlopt)                :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+PRAXIS (from nlopt)                                    :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+Nelder-Mead simplex (from nlopt)                       :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+sbplx (from nlopt)                                     :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+MMA (Method of Moving Asymptotes) (from nlopt)         :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+CCSA (from nlopt)                                      :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+SLSQP (from nlopt)                                     :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+low-storage BFGS (from nlopt)                          :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+preconditioned truncated Newton (from nlopt)           :class:`pygmo.nlopt`                      S-CU            Exposed from C++
+shifted limited-memory variable-metric (from nlopt)    :class:`pygmo.nlopt`                      S-CU            Exposed from C++
```

Comment:
Missing the augmented Lagrangian algorithm. Also, BOBYQA, NEWUOA, PRAXIS, Nelder-Mead, sbplx do not handle constraints (only bound constraints). Same for BFGS, Newton and variable-metric.